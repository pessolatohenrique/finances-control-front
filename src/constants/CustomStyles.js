import { styled } from "@mui/material/styles";
import { tableCellClasses } from "@mui/material/TableCell";
import { TableCell } from "@mui/material";

export const StyledTableCell = styled(TableCell)(({ theme }) => ({
  [`&.${tableCellClasses.head}`]: {
    fontSize: "1.1em",
    fontWeight: 300,
  },
}));
