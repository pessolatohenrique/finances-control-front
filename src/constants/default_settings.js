export const PROJECT_NAME = "Project Name";
export const THEME_COLOR = "#1976d2";
export const SNACKBAR_DIRECTION = { vertical: "top", horizontal: "right" };
