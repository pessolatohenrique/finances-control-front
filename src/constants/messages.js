export const REQUIRED_MESSAGE = "Campo de preenchimento obrigat√≥rio";
export const USER_NOT_EXISTS_MESSAGE = "This user doesnt exists";
export const WRONG_PASSWORD_MESSAGE = "Invalid username or password";
