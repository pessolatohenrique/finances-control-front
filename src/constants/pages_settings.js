export const PATHS_WITHOUT_LOGIN = ["/usuario/novo"];
