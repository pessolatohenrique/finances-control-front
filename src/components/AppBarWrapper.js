import React, { useState } from "react";
import {
  Typography,
  AppBar,
  Box,
  Toolbar,
  Button,
  IconButton,
  Drawer,
  List,
  ListItem,
  ListItemIcon,
} from "@mui/material";
import ChevronLeftIcon from "@material-ui/icons/ChevronLeft";
import MenuIcon from "@material-ui/icons/Menu";
import DashboardIcon from "@material-ui/icons/Dashboard";
import AttachMoneyIcon from "@material-ui/icons/AttachMoney";
import MoneyOffIcon from "@material-ui/icons/MoneyOff";
import { PROJECT_NAME } from "../constants/default_settings";
import { isPageWithoutLogin } from "../utils/pages";

const items = [
  {
    table: "dashboard",
    link: "",
    icon: <DashboardIcon />,
  },
  {
    table: "novo ganho",
    link: "ganho/novo",
    icon: <AttachMoneyIcon />,
  },
  {
    table: "nova despesa",
    link: "gasto/novo",
    icon: <MoneyOffIcon />,
  },
];

function AppBarWrapper(props) {
  const [open, setOpen] = useState(false);

  function handleDrawerOpen() {
    setOpen(true);
  }

  function handleDrawerClose() {
    setOpen(false);
  }

  function logout() {
    localStorage.removeItem("token");
    localStorage.removeItem("refreshToken");
    window.location.href = "/signin";
  }

  return (
    <Box sx={{ flexGrow: 1 }}>
      <AppBar position="static">
        <Toolbar>
          <IconButton
            size="large"
            edge="start"
            color="inherit"
            aria-label="menu"
            sx={{ mr: 2 }}
            data-testid="menu-button"
            onClick={handleDrawerOpen}
          >
            <MenuIcon />
          </IconButton>
          <Typography
            variant="h6"
            component="div"
            sx={{ flexGrow: 1 }}
            data-testid="project-name"
          >
            {PROJECT_NAME}
          </Typography>
          {!isPageWithoutLogin() && (
            <Button
              color="inherit"
              onClick={logout}
              data-testid="logout-button"
            >
              Logout
            </Button>
          )}
        </Toolbar>
      </AppBar>
      <Drawer variant="persistent" anchor="left" open={open}>
        <div>
          <IconButton onClick={handleDrawerClose}>
            <ChevronLeftIcon />
          </IconButton>
          <List>
            {items.map((item, key) => (
              <ListItem button key={key}>
                <ListItemIcon>{item.icon}</ListItemIcon>
                <Button size="small" color="primary" href={`/${item.link}`}>
                  {item.table}
                </Button>
              </ListItem>
            ))}
          </List>
        </div>
      </Drawer>
    </Box>
  );
}

export default AppBarWrapper;
