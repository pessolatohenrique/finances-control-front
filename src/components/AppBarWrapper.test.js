import { render, screen, fireEvent } from "@testing-library/react";
import AppBarWrapper from "./AppBarWrapper";
import { PROJECT_NAME } from "../constants/default_settings";

describe("AppBarWrapper", () => {
  it("should generate snapshot", () => {
    const { container } = render(<AppBarWrapper />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render app name", () => {
    render(<AppBarWrapper />);
    expect(screen.getByTestId("project-name")).toHaveTextContent(PROJECT_NAME);
  });

  it("should open menu", () => {
    const { container } = render(<AppBarWrapper />);
    const menuBtn = screen.getByTestId("menu-button");
    fireEvent.click(menuBtn);
    expect(container.getElementsByClassName("MuiList-root")).toHaveLength(1);
  });
});
