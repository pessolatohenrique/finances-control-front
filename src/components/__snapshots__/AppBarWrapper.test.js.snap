// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBarWrapper should generate snapshot 1`] = `
<div
  class="MuiBox-root css-i9gxme"
>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
    >
      <button
        aria-label="menu"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-ancrnh-MuiButtonBase-root-MuiIconButton-root"
        data-testid="menu-button"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="MenuIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <div
        class="MuiTypography-root MuiTypography-h6 css-1juivf6-MuiTypography-root"
        data-testid="project-name"
      >
        Finances Control
      </div>
      <button
        class="MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButtonBase-root  css-1y942vo-MuiButtonBase-root-MuiButton-root"
        data-testid="logout-button"
        tabindex="0"
        type="button"
      >
        Logout
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </header>
  <div
    class="MuiDrawer-root MuiDrawer-docked css-7ik032-MuiDrawer-docked"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-12i7wg6-MuiPaper-root-MuiDrawer-paper"
      style="visibility: hidden; webkit-transform: translateX(-0px); transform: translateX(-0px);"
    >
      <div>
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ChevronLeftIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <ul
          class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
        >
          <div
            class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button css-bshv44-MuiButtonBase-root-MuiListItem-root"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="PeopleIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16 11c1.66 0 2.99-1.34 2.99-3S17.66 5 16 5c-1.66 0-3 1.34-3 3s1.34 3 3 3zm-8 0c1.66 0 2.99-1.34 2.99-3S9.66 5 8 5C6.34 5 5 6.34 5 8s1.34 3 3 3zm0 2c-2.33 0-7 1.17-7 3.5V19h14v-2.5c0-2.33-4.67-3.5-7-3.5zm8 0c-.29 0-.62.02-.97.05 1.16.84 1.97 1.97 1.97 3.45V19h6v-2.5c0-2.33-4.67-3.5-7-3.5z"
                />
              </svg>
            </div>
            <a
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-1knaqv7-MuiButtonBase-root-MuiButton-root"
              href="/autores"
              tabindex="0"
            >
              autores
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </div>
          <div
            class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button css-bshv44-MuiButtonBase-root-MuiListItem-root"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="BookIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18 2H6c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zM6 4h5v8l-2.5-1.5L6 12V4z"
                />
              </svg>
            </div>
            <a
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-1knaqv7-MuiButtonBase-root-MuiButton-root"
              href="/livros"
              tabindex="0"
            >
              livros
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </div>
          <div
            class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button css-bshv44-MuiButtonBase-root-MuiListItem-root"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="BookIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18 2H6c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zM6 4h5v8l-2.5-1.5L6 12V4z"
                />
              </svg>
            </div>
            <a
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-1knaqv7-MuiButtonBase-root-MuiButton-root"
              href="/livros/5"
              tabindex="0"
            >
              livros detalhes
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </div>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
